ccm_web/server/src/canvas/canvas.service.ts

```diff
@@ -14,47 +14,62 @@ import { User } from '../user/user.model'
 import { CanvasConfig } from '../config'
 import { DatabaseError } from '../errors'
 import baseLogger from '../logger'
-import { IncomingHttpHeaders } from 'http'
 
 const logger = baseLogger.child({ filePath: __filename })
 
 type SupportedAPIEndpoint = '/api/v1/' | '/api/graphql/'
 
-interface IncomingRateLimitedCanvasHttpHeaders extends IncomingHttpHeaders {
-  'x-rate-limit-remaining'?: string
-  'x-request-cost'?: string
-}
-
 const requestorOptions: GotOptions = {
   retry: {
-    limit: 2,
+    limit: 4,
     methods: ['POST', 'GET', 'PUT', 'DELETE'],
-    statusCodes: got.defaults.options.retry.statusCodes.concat([403])
+    statusCodes: got.defaults.options.retry.statusCodes.concat([403]),
+    calculateDelay: ({ attemptCount, retryOptions, error, computedValue }) => {
+      const isStrangeError = !(error instanceof HTTPError) && error.code === undefined
+      const delay = computedValue === 0 && !isStrangeError ? 0 : 5000
```

Comment:
…this number of delay milliseconds (5000), probably should be parameterized values that are loaded from the application configuration.

I looked into doing that before my last commit.  It isn't **_very_** complex to support config parameters, but it does involve a lot of changes, especially in this file.  It was getting complicated enough that I decided it should wait until after we have a solution to the rate‑limit problem.

Reply:
I'm fine waiting on the configuration thing; it may be nice but also seems potentially unnecessary. It would require some changes to the service itself, probably you'd want to wrap the options creation in a function and pass the necessary configuration when the `Requestor` is instantiated.